kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -117,6 +120,13 @@ public TransactionBuilder withMaxRetries(int maxRetries) {
     return this;
   }
 
+  @Override
```

Comment:
This feels like a really weird config to be on the transaction builder since it doesn't apply to every transaction. Is there really no table property for this?

Reply:
Maybe this is another scenario where it would make sense for us to have sort of session config.. (and we're kind of bypassing it by putting it here)

Reply:
Also slightly confusing semantics about when we compact since it's writer based (and not required), but there's probably nothing we can do about that. But it's kind of confusing, do we compact when it's been logCompactionInterval commits since the last compaction? Since the last time this writer compacted based on this writer's number of commits? Based on mod (how it's implemented)?

Not sure if it's worth clarifying this to the user

Reply:
It probably should/could be an actual config on the _table_, but that would be a bigger change. Do we have any notion of a "session config"? (afaict, we don't). 

Unless there's a current place it fits better I'd advocate for having it here and then we can move it if we introduce something better?

Reply:
I think we've just been putting off adding something like a session config for a while. I guess this is another example of where it would make sense to have one

Reply:
Makes sense. Are you okay with that being a follow-up and having this here?

Reply:
My 2c is: yes that should be a followup. That's quite a large change (designing a configuration API and then plumbing it through everywhere). All of these APIs are maked as `@Evolving` -- so, IMO, it's fine to add this for now and refactor later 👍 Also curious what Alli things here, too.

Reply:
I'm okay with this being a follow up task -- just wanted to note this is another scenario where it seems like we should have a configuration (and curious if others agree)